docs/website/docs/user-guides/advanced/connecting-a-service.md

```diff
@@ -0,0 +1,229 @@
+---
+title: Connecting to a service
+sidebar_position: 1
+---
+
+This tutorial will show you how you can connect your Go application to a mongodb service.
+
+Building on top of the [Go quickstart guide](../quickstart/go.md), we will extend the application to check if it is connected to a mongodb service.
+
+### Pre-requisites
+1. Install Service Binding Operator on your cluster.  See [installation instruction](https://operatorhub.io/operator/service-binding-operator).
+
+2. Install Percona Server Mongodb operator on your cluster. See [installation instruction](https://operatorhub.io/operator/percona-server-mongodb-operator).
+:::note
+The operator will be installed in a new namespace called "my-percona-server-mongodb-operator" and will be usable from that namespace only.
+:::
+3. Create a mongodb service.
+
+import CreateMongodbService from './_create-mongodb-service.mdx';
+
+<CreateMongodbService/>
+
+### Implement the code logic
+:::note
+If you're already running `odo dev` in a terminal, exit it and start afresh.
+:::
+
+The new code is simple. We obtain the connection information (username, password, and host) from the environment, and use it to connect to the mongodb service and ping it. 
+
+Replace the content of your `main.go` with the following content:
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+// Connection URI
+var (
+	USERNAME = os.Getenv("PERCONASERVERMONGODB_MONGODB_USER_ADMIN_USER")
```

Comment:
We should add to the tutorial how these environment variables were obtained. Isn't it shown with `odo describe`?

Reply:
`odo describe binding` also exposes a lot of other environment variables. I think it will be simpler to not talk about them for the sake of simplicity.

Reply:
Even I didn't realize that it actually shows the environment variable.. how could a new user know either when this doc is in an initial user guide? 

A user can install mongodb operator and then realize after "what are the env variables again I should use?"

I think it's important to describe *before* this step that we should explain how these environment variables are retrieved / added under prerequisites.

Example, for Step 4 in prerequisites, we should describe how we are able to find out all these variables that can be used in your application.